manifests/server/config.pp

```diff
@@ -29,16 +29,35 @@
     mode   => '0755',
   }
 
-  file { $cfg_file:
-    ensure  => present,
-    owner   => $owner,
-    group   => $group,
-    mode    => '0644',
-    require => [
-      File[$cfg_dir],
-      Class['dns::server::install']
-    ],
-    notify  => Class['dns::server::service'],
+  case $::osfamily {
+    'RedHat': {
+      file { $cfg_file:
+        ensure  => present,
+        owner   => $owner,
+        group   => $group,
+        mode    => '0644',
+        content => template("${module_name}/named.conf.erb"),
```

Comment:
complete diff of changes to make your fork pass spec:

    diff --git a/spec/classes/dns__server__config_spec.rb b/spec/classes/dns__server__config_spec.rb
    index fc386e8..cc6ce82 100644
    --- a/spec/classes/dns__server__config_spec.rb
    +++ b/spec/classes/dns__server__config_spec.rb
    @@ -35,8 +35,8 @@ describe 'dns::server::config', :type => :class do
           'owner'  => 'named',
           'group'  => 'named',
           'mode'   => '0644',
    -    })
    -    it shoud contain_file('/etc/named.conf').with_content("/^include /etc/named/named.conf.options;$/")
    +    })}
    +    it { should contain_file('/etc/named.conf').with_content(/^include "\/etc\/named\/named.conf.options";$/) }
       end
     end

    diff --git a/templates/named.conf.erb b/templates/named.conf.erb
    index 3221629..10ffbb9 100644
    --- a/templates/named.conf.erb
    +++ b/templates/named.conf.erb
    @@ -1,7 +1,7 @@
     ; File managed by puppet.
     ;

    -include /etc/named/named.conf.options;
    +include "/etc/named/named.conf.options";
     logging {
             channel default_debug {
                     file "data/named.run";

I added the `"…"` around the `named.conf.options` include to match correct `named.conf` syntax.